robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -1,191 +1,545 @@
+        antennaHandlerTemplate('lat', 'DMS')
+    }
+
+    function createAntennaLongitudeHandler() {
+        antennaHandlerTemplate('long', 'DD')
+        antennaHandlerTemplate('long', 'DDM')
+        antennaHandlerTemplate('long', 'DMS')
+    }
+
+    $('#toggle-goal-modal-btn').on('click', function(event) {
+        $("#goal-modal-body-content").empty()
+        let goalTemplate = $('#created-goal-template').html()
+        for (i = 0; i < goalCount; i++) {
+            $("#goal-modal-body-content").append(goalTemplate)
+            $("#goal-modal-body-content .goal").addClass('goal-' + i).removeClass('goal')
+            $(".goal-" + i).find('*').addClass('goal-' + i)
+
+            createGoalButtons(i)
+
+            $('#goal-name.goal-' + i).val(goalList[i].name)
+            $('#goal-confirm-btn.goal-' + i).prop('disabled', true)
+            $('#goal-change-btn.goal-' + i).prop('disabled', false)
+            $('div.goal-' + i + ' fieldset').prop('disabled', true)
+            $('#goal-DD-dec-deg-input.lat.goal-' + i).val(goalList[i].lat)
+            $('#goal-DD-dec-deg-input.long.goal-' + i).val(goalList[i].long)
+
+        }
+    })
+
+    $("#new-goal-coordinates-btn").on('click', function(event) {
+        let newGoalTemplate = $('#add-goal-coordinates').html()
+
+        $("#goal-modal-body-content").append(newGoalTemplate)
+        $("#goal-modal-body-content .goal").addClass('goal-' + goalCount).removeClass('goal')
+        $(".goal-" + goalCount).find('*').addClass('goal-' + goalCount)
+
+        createGoalLatitudeHandler(goalCount)
+        createGoalLongitudeHandler(goalCount)
+        createGoalButtons(goalCount)
+    })
+
+    function saveGoalData(current, lat_format, long_format) {
+        let lat = null
+        let long = null
+        try {
+            for (i = 0; i < goalCount; i++) {
+                if ($('#goal-name.goal-' + current).val() == goalList[i].name) {
+                    throw 'Enter different name'
+                }
+            }
+            if (lat_format == 0) {
+                //decimal-degrees mode
+                decdeg = parseFloat($('#goal-DD-dec-deg-input.lat.goal-' + current).val())
+
+                if (isNaN(decdeg)) throw "Bad input lat DD"
+
+                $('#goal-DD-dec-deg-input.lat.goal-' + current).attr("value", decdeg)
+
+                lat = decdeg
+            } else if (lat_format == 1) {
+                //degrees-decimal-minutes mode
+                let dec = parseFloat($('#goal-DDM-deg-input.lat.goal-' + current).val())
+                let min = parseFloat($('#goal-DDM-dec-min-input.lat.goal-' + current).val())
+
+                if (isNaN(dec) || isNaN(min)) throw "Bad input lat DDM"
+
+                lat = dec + (min / 60)
+
+                $('#goal-DDM-deg-input.lat.goal-' + current).attr("value", dec)
+                $('#goal-DDM-dec-min-input.lat.goal-' + current).attr("value", min)
+
+            } else if (lat_format == 2) {
```

Comment:
These magic vars (0, 1, 2) should be declared as variables.
Further more, 60 and 3600 should be handled the same way.

Reply:
Fixed